core/src/main/java/feign/Response.java

```diff
@@ -314,7 +320,7 @@ public Reader asReader() {
     }
 
     @Override
-    public Reader asReader(Charset charset) throws IOException {
```

Comment:
Can we undo these `throws IOException`?

My concern is breaking existing feign users

Reply:
this is just on the implementation which is private final.
see above:

`private static final class InputStreamBody implements Response.Body`
so no feign user would be able to override this. and the exception is not thrown